TreasureData/TreasureData.h

```diff
@@ -11,41 +11,40 @@
 
 @interface TreasureData : NSObject
 
-@property(nonatomic, strong) TDClient *client;
+@property(nonatomic, strong) TDClient * _Nullable client;
 
-@property(nonatomic, strong) NSString *defaultDatabase;
+@property(nonatomic, strong) NSString * _Nullable defaultDatabase;
 
-+ (void)initializeWithApiKey:(NSString *)apiKey;
++ (void)initializeWithApiKey:(NSString * _Nonnull)apiKey;
 
-+ (instancetype)sharedInstance;
+// Can not be null after initializeWithApiKey: has been called.
++ (instancetype _Nonnull)sharedInstance;
 
-+ (void)initializeApiEndpoint:(NSString *)apiEndpoint;
++ (void)initializeApiEndpoint:(NSString * _Nullable)apiEndpoint;
```

Comment:
Would it be better if we also enforces a Nonnull value here? I think there is no reason one would want to explicitly set a nil endpoint.

Reply:
@huylenq 
Thanks for checking my PR!

I originally used _Nonnull but decided to change to _Nullable because:
- initializeApiEndpoint updates the static defaultApiEndpoint.
- defaultApiEndpoint == nil means "https://in.treasuredata.com" will be used.
- It might be that someone wants to release one TreasureData instance, switch back to the default endpoint, and initialize a new instance.

But I don't know if the above is really something necessary to support or intended usage...

I can change to _Nonnull if that's what you prefer.

Reply:
That's an arbitrary use case but fair points. Thanks @nevil